includes/admin/feedzy-rss-feeds-admin.php

```diff
@@ -2371,4 +2370,30 @@ private function enable_telemetry() {
 		add_filter( 'themeisle_sdk_enable_telemetry', '__return_true' );
 		wp_enqueue_script( $this->plugin_name . '_telemetry', FEEDZY_ABSURL . 'js/telemetry.js', array(), $this->version, true );
 	}
+
+	/**
+	 * Set the black friday data.
+	 *
+	 * @param array $config The configuration array.
+	 * @return array
+	 */
+	public function set_black_friday_data( $config ) {
+		$product_label = __( 'Feedzy RSS Feeds', 'feedzy-rss-feeds' );
+		$discount      = '40%';
+
+		if ( feedzy_is_pro() ) {
+			$product_label = __( 'Feedzy RSS Feeds Pro', 'feedzy-rss-feeds' );
+			$discount = '50%';
+		}
+
+		$license_data = get_option( 'feedzy_rss_feeds_pro_license_data', array() );
+
+		$config['message'] = sprintf( __( 'Our biggest sale of the year: <strong>%1$s OFF</strong> on <strong>%2$s</strong>! Don\'t miss this limited-time offer.', 'feedzy-rss-feeds' ), $discount, $product_label );
+		$config['url']    = add_query_arg( array(
+			'utm_plugin' => feedzy_is_pro() ? 'feedzy-pro' : 'feedzy-rss-feeds',
+			'utm_plan'   => self::plan_category( $license_data ),
+		), $config['base_url'] );
```

Comment:
Let's use here  

```php
add_query_arg(
	[
		'utm_term' => feedzy_is_pro() ? 'plan-' . apply_filters( 'product_feedzy_license_plan', 0 ) : 'free'
	],
	tsdk_translate_link( tsdk_utmify( 'https://themeisle.com/plugins/feedzy-rss-feeds/blackfriday', 'bfcm', 'feedzy' ) ) );

```

Reply:
@Soare-Robert-Daniel lets also include the licensekey as lkey using product_feedzy_license_key

Reply:
good point, we can. Lets do this in the SDK, by using remove_query_arg if the url contains lkey and the user does not have the right permision.

Reply:
With [this](https://github.com/Codeinwp/themeisle-sdk-main/pull/295#issuecomment-2875465003) we have available the `$current_product`, should we use this to get the plan (priceID) and the license key directly from the SDK using the `product_<slug>` filters instead of having it in each product?